src/pages/index.js

```diff
@@ -52,36 +123,90 @@ function createCardObject() {
 const popupEditProfile = new PopupWithForm({
   popupSelector: "#modal_one",
   handleFormSubmit: (formData) => {
-    userInfo.setUserInfo(formData);
-    popupEditProfile.close();
-    popupEditProfile.formElement.reset();
+    return api
+      .updateUserInfo({ userName: formData.name, about: formData.job })
+      .then((updatedUserData) => {
+        userInfo.setUserInfo({
+          name: updatedUserData.name,
+          job: updatedUserData.about,
+        });
+      })
+      .catch((err) => {
+        console.error(`Failed to update user data: ${err}`);
+      });
   },
 });
 
-// Popup for adding new cards
+// Popup for adding new cards old
+// const popupAddCard = new PopupWithForm({
+//   popupSelector: "#modal_adding-cards",
+//   handleFormSubmit: (formData) => {
+//     const cardData = { name: formData.title, link: formData.url };
+//     renderCard(cardData);
+//     cardFormValidator.disableBtn();
+//     popupAddCard.close();
+//     popupAddCard.formElement.reset();
+//   },
+// });
+
 const popupAddCard = new PopupWithForm({
   popupSelector: "#modal_adding-cards",
   handleFormSubmit: (formData) => {
-    const cardData = { name: formData.title, link: formData.url };
-    renderCard(cardData);
-    cardFormValidator.disableBtn();
-    popupAddCard.close();
-    popupAddCard.formElement.reset();
+    return api
+      .addNewCard({ locationName: formData.title, link: formData.url })
+      .then((newCardData) => {
+        renderCard(newCardData); // Add the newly created card to the UI
+        cardFormValidator.disableBtn();
+      })
+      .catch((err) => {
+        console.error(`Failed to add a new card: ${err}`);
+      });
   },
 });
 
+function handleLikeClick(cardData, card) {
+  if (cardData.isLiked === false) {
+    api.likeCard(cardData._id).then((newCardData) => {
+      card.updateLikeBtn(newCardData.isLiked);
+    });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**